subtree_compare.cpp

```diff
@@ -0,0 +1,150 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include <iostream>
+
+using namespace std;
+
+// basic node struct used in tree
+struct node {
+	node * left;
+	node * right;
+
+	int value;
+
+	node( int val ) : value(val), left(NULL), right(NULL) {}
+	node( int val, node * l, node * r ) : value(val), left(l), right(r) {}
+};
+
+// basic struct for holding tree
+struct Tree {
+	
+	node * root;
+	
+	Tree() : root(NULL) {}
+	void populate( int * num_arr, int n);
+};
+
+// recursively compares tree sub branches
+bool compare( node * t, node * s ) {
+	if ( t == NULL && s == NULL ) 
+		return true;
+	if( (t == NULL && s != NULL) || (t != NULL && s == NULL) )
+		return false;
+	
+	if ( t->value == s->value )
+		return compare( t->left, s->left ) && compare( t->right, s->right );
+	else 
+		return false;
+}
+
+// build binary tree from list of numbers
+void Tree::populate( int * num_arr, int n ) {
```

Comment:
Ideally you should keep these in separate files/classes, as these utilities may be useful for other programs you write, and you don't want to duplicate this every time you have to solve a tree problem